Use timeout_in_minutes for Terraform timeout in AWS CloudFormation

andrewtarry opened the pull request:
We recently encountered a problem with the CloudFormation resource due to the timeout_in_minutes value being increased. If the CloudFormation timeout is increased Terraform will timeout after 30 minutes resulting an error and other resources not being applied but CloudFormation carries on building.

This change leave the default timeout at 30 minutes but if the timeout_in_minutes is greater than 30 the timeout becomes timeout_in_minutes + 5 minutes (just to make sure it will have finished or failed)

radeksimko commented on builtin/providers/aws/resource_aws_cloudformation_stack.go:
```diff
@@ -143,7 +149,7 @@ func resourceAwsCloudFormationStackCreate(d *schema.ResourceData, meta interface
 		Pending:    []string{"CREATE_IN_PROGRESS", "ROLLBACK_IN_PROGRESS", "ROLLBACK_COMPLETE"},
 		Target:     []string{"CREATE_COMPLETE"},
-		Timeout:    30 * time.Minute,
+		Timeout:    time.Duration(int64(timeout)) * time.Minute,
```
It would have been cleaner to start with `int64` variable instead of casting it later.

Not that I would be worried about _minutes_ going over 32bits, but strictly speaking we might potentially end up comparing int32 with int64 in the condition above.

i.e. this 
```go
timeout := 30
```
can be turned into
```go
retryTimeout := int64(30)
```
and the line here can be turned into 
```go
Timeout:    time.Duration(retryTimeout) * time.Minute,
```

--- 

as hinted above, I'd also suggest renaming the `timeout` variable to more specific `retryTimeout`.

radeksimko commented:
Hi @andrewtarry 
firstly thanks for discovering this edge case and taking the time to submit the PR! :+1: 

I left you one comment there regarding data type and variable name.

One last thing, before we merge this (and I'm aware I might be asking for too much - depending on your git skills) - would you mind squashing all 3 commits into a single one using `git rebase -i HEAD~3` and `git push -f`? If you're too worried about this, I can do it for you as part of the merging process, but it would be very helpful if you have done that in your fork, so I can just push the green merge button. :smiley:

andrewtarry commented:
Hi @radeksimko 

Thanks for your comments.

I have removed the extra casting as you suggesting and rebased the commits. Travis seems to be failing but it looks like it's a issue between Travis and github.
Thanks

radeksimko commented:
@andrewtarry FYI for the future, it might make your life easier if you create a PR from a feature/bugfix branch, not from `master`. Generally _force-pushing_ `master` (even if it's just fork) is not considered a best practice and it makes it difficult to send+maintain more PRs to the same repo. :wink:
